src/components/templates/TemplateWorkflowsContent.vue

```diff
@@ -1,68 +1,134 @@
+                />
+                <a>
+                  <div
+                    class="absolute top-0 left-0 w-64 h-64 overflow-hidden opacity-0 transition duration-300 ease-in-out hover:opacity-100 bg-opacity-50 bg-black flex items-center justify-center"
+                  >
+                    <i class="pi pi-play-circle" style="color: white"></i>
+                  </div>
+                </a>
+                <ProgressSpinner
+                  v-if="loading === slotProps.data"
+                  class="absolute inset-0 z-1 w-3/12 h-full"
+                />
               </div>
-            </a>
-            <ProgressSpinner
-              v-if="loading === template"
-              class="absolute inset-0 z-1 w-3/12 h-full"
-            />
-          </div>
-        </template>
-        <template #subtitle>{{
-          $t(`templateWorkflows.template.${template}`)
-        }}</template>
-      </Card>
-    </div>
+            </div>
+          </template>
+          <template #subtitle>{{ slotProps.data }}</template>
+        </Card>
+      </template>
+    </Carousel>
   </div>
 </template>
 
 <script setup lang="ts">
-import { useDialogStore } from '@/stores/dialogStore'
+import Carousel from 'primevue/carousel'
+import Listbox from 'primevue/listbox'
 import Card from 'primevue/card'
 import ProgressSpinner from 'primevue/progressspinner'
-import { ref } from 'vue'
+import { useToast } from 'primevue/usetoast'
+import { onMounted, ref } from 'vue'
+import { useDialogStore } from '@/stores/dialogStore'
 import { app } from '@/scripts/app'
 import { api } from '@/scripts/api'
 import { useI18n } from 'vue-i18n'
+
+interface WorkflowTemplatesTab {
+  moduleName: string
+  title: string
+  templates: string[]
+}
+
 const { t } = useI18n()
+const toast = useToast()
+
+//These default templates are provided by the frontend
+const comfyUITemplates: WorkflowTemplatesTab = {
+  moduleName: 'default',
+  title: 'ComfyUI',
+  templates: ['default', 'image2image', 'upscale', 'flux_schnell']
+}
 
-const templates = ['default', 'image2image', 'upscale', 'flux_schnell']
+const tabs = ref<WorkflowTemplatesTab[]>([comfyUITemplates])
+const selectedTab = ref<WorkflowTemplatesTab>(comfyUITemplates)
 const loading = ref<string | null>(null)
 
+onMounted(async () => {
+  try {
+    const workflowTemplates = await api.getWorkflowTemplates()
+    tabs.value = [
+      comfyUITemplates,
+      ...Object.entries(workflowTemplates).map(([key, value]) => ({
+        moduleName: key,
+        title: key,
+        templates: value
+      }))
+    ]
+  } catch (error) {
+    console.error('Error fetching workflow templates:', error)
+    toast.add({
+      severity: 'error',
+      summary: 'Error',
+      detail: 'Failed to fetch workflow templates',
+      life: 5000
+    })
+  }
+})
+
 const loadWorkflow = async (id: string) => {
   loading.value = id
-  const json = await fetch(api.fileURL(`templates/${id}.json`)).then((r) =>
-    r.json()
-  )
+  let json
```

Comment:
Let's extract the storage of custom nodes workflow templates as a Pinia store to avoid reloading everytime the dialog is opened.